odoo/addons/test_new_api/tests/test_new_fields.py

```diff
@@ -1280,6 +1280,99 @@ def test_27_company_dependent(self):
+        self.assertEqual(Model.search([('foo', '=', False)]), Model.browse())
+        self.assertEqual(Model.search([('foo', '!=', False)]), record_empty | record_qwer | record_azer)
+
+    def test_28_company_dependent_search_many2one(self):
+        Model = self.env['test_new_api.company']
+        TagModel = self.env['test_new_api.multi.tag']
+        normal_tag = TagModel.create({'name': 'normal'})
+        urgent_tag = TagModel.create({'name': 'urgent'})
+
+        record_empty = Model.create({})
+        record_normal = Model.create({'tag_id': normal_tag.id})
+        record_urgent = Model.create({'tag_id': urgent_tag.id})
+
+        # Without default values
+        self.assertEqual(Model.search([('tag_id', 'like', 'normal')]), record_normal)
+        self.assertEqual(Model.search([('tag_id', 'not like', 'normal')]), record_empty | record_urgent)
+        self.assertEqual(Model.search([('tag_id', '=', False)]), record_empty)
+        self.assertEqual(Model.search([('tag_id', '!=', False)]), record_normal | record_urgent)
+
+        field_id = self.env['ir.model.fields'].search([('model', '=', 'test_new_api.company'), ('name', '=', 'tag_id')], limit=1)
+        self.env['ir.property'].create({
+            'name': 'tag_id',
+            'company_id': self.env.company.id,
+            'fields_id': field_id.id,
+            'value_reference': f'test_new_api.multi.tag,{normal_tag.id}',
+            'type': 'many2one',
+        })
+        self.assertEqual(record_empty.tag_id, normal_tag)
+        self.assertEqual(Model.search([('tag_id', 'like', 'norm')]), record_empty | record_normal)
+        self.assertEqual(Model.search([('tag_id', 'not like', 'norm')]), record_urgent)
+        self.assertEqual(Model.search([('tag_id', '=', False)]), Model.browse())
+        self.assertEqual(Model.search([('tag_id', '!=', False)]), record_empty | record_normal | record_urgent)
+
+    def test_29_company_dependent_search_boolean(self):
+        Model = self.env['test_new_api.company']
+
+        record_empty = Model.create({})
+        record_true = Model.create({'truth': True})
+        record_false = Model.create({'truth': False})
+        # Without default values
+        self.assertEqual(Model.search([('truth', '=', True)]), record_true)
+        self.assertEqual(Model.search([('truth', '!=', False)]), record_true)
+        self.assertEqual(Model.search([('truth', '!=', True)]), record_empty | record_false)
+        self.assertEqual(Model.search([('truth', '=', False)]), record_empty | record_false)
+
+        field_id = self.env['ir.model.fields'].search([('model', '=', 'test_new_api.company'), ('name', '=', 'truth')], limit=1)
+        pro = self.env['ir.property'].create({
+            'name': 'truth',
+            'company_id': self.env.company.id,
+            'fields_id': field_id.id,
+            'value_integer': 0,  # For false
+            'type': 'boolean',
+        })
```

Comment:
There is an API for that:
```suggestion
        self.env['ir.property']._set_default('truth', 'test_new_api.company', False, self.env.company)
```

Reply:
don't exist in 13.0 :'(

Reply:
Ah mince alors, c'est juste!